ts/features/fci/screens/FciRouterScreen.tsx

```diff
@@ -37,25 +47,56 @@ const FciSignatureScreen = (
     />
   );
 
-  const ErrorComponent = () => (
-    <GenericErrorComponent
-      title={I18n.t("features.fci.errors.generic.default.title")}
-      subTitle={I18n.t("features.fci.errors.generic.default.subTitle")}
-      onPress={() => dispatch(fciEndRequest())}
-      testID="GenericErrorComponentTestID"
-    />
-  );
+  const GenericError = (status?: ProblemJson["status"]) => {
+    if (status === 404) {
+      return (
+        <ErrorComponent
+          title={I18n.t("features.fci.errors.generic.wrongUser.title")}
+          subTitle={I18n.t("features.fci.errors.generic.wrongUser.subTitle")}
+          image={doubt}
+          onPress={() => dispatch(fciEndRequest())}
+          testID="WrongUserErrorComponentTestID"
+        />
+      );
+    }
+
+    return (
+      <GenericErrorComponent
+        title={I18n.t("features.fci.errors.generic.default.title")}
+        subTitle={I18n.t("features.fci.errors.generic.default.subTitle")}
+        onPress={() => dispatch(fciEndRequest())}
+        testID="GenericErrorComponentTestID"
+      />
+    );
+  };
+
+  const renderErrorComponent = (error?: NetworkError) =>
+    pipe(
+      error,
+      O.fromNullable,
+      O.map(e => getErrorFromNetworkError(e)),
+      O.map(error => getGenericError(error)),
+      O.map(error => JSON.parse(error.value.message)),
+      O.map(ProblemJson.decode),
+      O.map(
+        E.fold(
+          () => GenericError(),
+          problemJson => GenericError(problemJson.status)
+        )
+      ),
+      O.getOrElse(() => GenericError())
+    );
 
   return pot.fold(
     fciSignatureRequest,
     () => <LoadingComponent />,
     () => <LoadingComponent />,
     () => <LoadingComponent />,
-    _ => <ErrorComponent />,
+    renderErrorComponent,
     b => <SuccessComponent signatureRequest={b} />,
     () => <LoadingComponent />,
     () => <LoadingComponent />,
-    _ => <ErrorComponent />
+    () => renderErrorComponent()
```

Comment:
```suggestion
    renderErrorComponent
```

can we use the same call as line 95?

Reply:
It returns a type error related to pot.foldSomeError that should be a (value, error) instead in line 95 it's a foldNoneError. Maybe the renderErrorComponent could be refactored to have as input (value?, error?). What do you think?
`const renderErrorComponent = (value?: any, error?: NetworkError) =>`